### ISSUE TITLE
v3.14.0 deprecates single hyphen for concise line

### ISSUE THREAD

**Issue by albertogasparin** [marko-js/marko#498]
### Context
Updating from version v3.13.0 to v3.14.0 adds lots of warnings about single hyphen for concise line being deprecated. I think such behaviour should occur on next major release as it requires a lot of code changes. Moreover, there is no clear documentation on how code should be changed as replacing `-` with `--` does not always make the warning disappear.

```
The usage of a single hyphen at the start of a concise line is now deprecated. Use "--" instead.
See: https://github.com/marko-js/htmljs-parser/issues/43
```

<details>

### Your Environment

* Version used: v3.14.0
* Environment name and version: node.js 6.9

### Code sample producing warnings

This is valid v3.13 syntax
``` marko
<!DOCTYPE html>
html [lang="en"]
  body
    h1 - Title
    script
      - var a = 'marko';
```

If translated to either this styles it still outputs the warning (for line 6):

``` marko
<!DOCTYPE html>
html [lang="en"]
  body
    h1 -- Title
    script
      -- var a = 'marko';
```

``` marko
<!DOCTYPE html>
html [lang="en"]
  body
    h1 -- Title
    script
      ---
      var a = 'marko';
      ---
```

Only this style is not printing any warning: 

``` marko
<!DOCTYPE html>
html [lang="en"]
  body
    h1 -- Title
    script --
       var a = 'marko';
```

So will this last one be the only allowed style in the future?

</details>

**Comment by patrick-steele-idem** [comment]
Hey @albertogasparin, it was necessary to deprecate a single hyphen in v3 because we have to remove support for single hyphens in v4 to support the improvement of allowing spaces in attribute values when the attribute value is a complex expression. If you are not able to get rid of the deprecation warning then that is a bug. I am investigating now and will publish a new version of `htmljs-parser` if necessary. I will update soon. Thanks for reporting the problem!

**Comment by patrick-steele-idem** [comment]
Sorry for the bad deprecation warning when using a double hyphen. The problem has been fixed with `htmljs-parser@2.2.2`. I know the deprecation warnings can be a lot of noise, but we didn't see a good way to avoid this change as part of Marko v4. I hope it is not too much of a pain to update your code now, but we do plan on providing a migration tool a little later that would automatically fix the issue, but it is not ready yet.

**Comment by albertogasparin** [comment]
Thanks for the quick resolution. As wrapping code with `---` is still valid I don't think it will be a major pain. 
I just had an "oh dear, WTF" moment after updating Marko, as nothing was said in the changelog so I thought npm went crazy and messed up versions 😉
